cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/ElasticsearchGenerator.scala

```diff
@@ -0,0 +1,430 @@
+      case Relation.== =>
+        s"""{"match": {"$fieldExpr": ${filter.values(0)}}}"""
+      case _ => throw new QueryParsingException("no supported parameter for this number in Elasticsearch")
+    }
+  }
+
+  private def parseTimeRelation(filter: FilterStatement, fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.inRange =>
+        s"""{"range": {"$fieldExpr": {"gte": "${filter.values(0)}", "lt": "${filter.values(1)}"}}}"""
+      case Relation.< =>
+        s"""{"range": {"$fieldExpr": {"lt": "${filter.values(0)}"}}}"""
+      case Relation.> =>
+        s"""{"range": {"$fieldExpr": {"gt": "${filter.values(0)}"}}}"""
+      case Relation.<= =>
+        s"""{"range": {"$fieldExpr": {"lte": "${filter.values(0)}"}}}"""
+      case Relation.>= =>
+        s"""{"range": {"$fieldExpr": {"gte": "${filter.values(0)}"}}}"""
+      case _ => throw new QueryParsingException("no supported parameter for this date in Elasticsearch")
+    }
+  }
+
+  private def parseStringRelation(filter: FilterStatement, fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.matches => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""{"match": {"$fieldExpr": "${values(0)}"}}"""
+      }
+      case Relation.!= => throw new QueryParsingException("filter relation: !=, not implemented in Elasticsearch adapter")
+      case Relation.contains => throw new QueryParsingException("filter relation: contains, not implemented in Elasticsearch adapter")
+    }
+  }
+
+  private def parseTextRelation(filter: FilterStatement, fieldExpr: String): String = {
+    val words = filter.values.map(w => s"${w.asInstanceOf[String].trim()}").mkString("\"", ",", "\"")
+    s"""{"match": {"${fieldExpr}": {"query": $words, "operator": "and"}}}"""
+  }
+
+  private def parseIntervalDuration(interval: Interval): String = {
+    import TimeUnit._
+    interval.unit match {
+      case Second => "second"
+      case Minute => "minute"
+      case Hour => "hour"
+      case Day => "day"
+      case Week => "week"
+      case Month => "month"
+      case Year => "year"
+    }
+  }
+
+  private def parseGroupByFunc(by: ByStatement, fieldExpr: String, groupStr: StringBuilder, isLookUp: Boolean, selectOpt: Option[SelectStatement]): Unit = {
+    by.funcOpt match {
+      case Some(func) =>
+        func match {
+          case bin: Bin => ???
+          case interval: Interval =>
+            val duration = parseIntervalDuration(interval)
+            groupStr.append(s""" "date_histogram": {"field": "${fieldExpr}", "interval": "$duration"} """.stripMargin)
+          case level: Level => {
+            val hierarchyField = by.field.asInstanceOf[HierarchyField]
+            val field = hierarchyField.levels.find(_._1 == level.levelTag).get._2
+            if (isLookUp) {
+              groupStr.append(s""" "terms": {"field": "${field}", "size": 2147483647} """.stripMargin)
```

Comment:
Same here: Or is there a way to configure this number inside Elastic instead of explicitly indicate it in every query? Seems redundant.

Reply:
You're right. Magic number is not a good coding style. I will fix it by using a constant variable.